### DISCUSSION TITLE
martin vs mapbox server

### DISCUSSION THREAD

**Discussion by amdevelopment-world** [maplibre/martin#739] [Q&A]
hi, i'm martin :)

i'm probably naive,i imported my data using the close to standard command:

`osm2pgsql --create --slim --number-processes 8 --keep-coastlines --flat-nodes=nodes --database osm -U postgres -H localhost -P 5433 europe-latest.osm.pbf
`
i was expecting to be able to run martin, and have the same result as when i connected to the mapbox server, the same look, everything.

but instead i'm getting error messages about missing layers etc.

i know, i hope that this is because of my lack of knowledge.

the whole point for me was to replace an expensive server, by my own, i see things don't really work this way.

the examples of how to use martin with maplibre are incomplete, not even a complete page of code + html
there is no explanation, or i could not find one, about how to use martin as a full fledge vector server for OSM

isn't that the reason of existence of this project ?

either i didn't find the explanation, because it's not visible enough, or i just wasted 10 days of work, by my fault.

in that case, can someone point me in the right direction, if any ?

my needs:
i want a vector tile based map (so i guess i need maplibre)
i need  a server that can produce those tiles

i don't even know where to start, because i don't see a confirmation anywhere that tells me this is even possible

**Comment by nyurik** [comment]
Martin is not exactly for that use case.  Instead, I highly recommend using [planetiler](https://github.com/onthegomap/planetiler) -- it can generate all your base tiles in just an hour on a single machine.   Once generated, Martin can serve those tiles directly from the planetiler-generated mbtiles, or you can generate pmtiles and put it on some s3/gcp bucket somewhere and not need any server at all.  Martin is is also useful if you have some extra data to overlay on top of the base map

**Comment by amdevelopment-world** [comment]
Thank you, that would work for me, a static base-map, and some dynamic content (colors), generated on the client side.

On 9 Jul 2023, at 05:39, Yuri Astrakhan ***@***.***> wrote:


Martin is not exactly for that use case. Instead, I highly recommend using planetiler <https://github.com/onthegomap/planetiler> -- it can generate all your base tiles in just an hour on a single machine. Once generated, Martin can serve those tiles directly from the planetiler-generated mbtiles, or you can generate pmtiles and put it on some s3/gcp bucket somewhere and not need any server at all. Martin is is also useful if you have some extra data to overlay on top of the base map
